### ISSUE TITLE
mlogfilter fails with AttributeError when using python3

### ISSUE THREAD

**Issue by udayjalagam** [rueckstiess/mtools#736]
Hello , 

I am trying to install mtools in python3 environment but it fails because of AttributeError when I try below command. 

> sudo tail -F /var/log/mongodb/mongodb.log | grep -i filemeta | tr -dc '\0-\177' | mlogfilter --slow 5000 

 
### Expected behavior
It supposed to display all the queries which are taking more than 5 seconds. 

### Actual/current behavior

It failed with below error . 

`Traceback (most recent call last):
  File "/usr/local/bin/mlogfilter", line 10, in <module>
    sys.exit(main())
  File "/opt/ns/diopython/lib/python3.5/site-packages/mtools/mlogfilter/mlogfilter.py", line 355, in main
    tool.run()
  File "/opt/ns/diopython/lib/python3.5/site-packages/mtools/mlogfilter/mlogfilter.py", line 332, in run
    for logevent in self.logfile_generator():
  File "/opt/ns/diopython/lib/python3.5/site-packages/mtools/mlogfilter/mlogfilter.py", line 231, in logfile_generator
    for logevent in self.args['logfile'][0]:
  File "/opt/ns/diopython/lib/python3.5/site-packages/mtools/util/logfile.py", line 248, in __iter__
    le = self.next()
  File "/opt/ns/diopython/lib/python3.5/site-packages/mtools/util/logfile.py", line 215, in next
    line = line.decode('utf-8', 'replace')
AttributeError: 'str' object has no attribute 'decode'`

### Steps to reproduce the actual/current behavior
```
virtualenv  myenv -p python3
source myenv/bin/activate
pip install mtools[all] 
sudo tail -F /var/log/mongodb/mongodb.log | grep -i filemeta | tr -dc '\0-\177' | <mk tools path >/mlogfilter --slow 5000
```

### Environment

| Software         | Version
| ---------------- | -------
| mtools           |   mtools-1.5.3
| MongoDB server   |   mongo 3.6.12
| Operating system | Ubuntu 16.06

`$ pip freeze
matplotlib==1.4.3
mtools==1.5.3
nose==1.3.7
numpy==1.14.5
pkg-resources==0.0.0
psutil==5.4.2
pymongo==3.6.1
pyparsing==2.4.2
python-dateutil==2.7.0
pytz==2019.2
six==1.12.0`

**Comment by udayjalagam** [comment]
The  issue is if you open file in read mode in python3 when reading it will automatically decode line but in code we are again trying to decode in mtools/util/logfile.py . below is the code snippet .  

```    def next(self):
        """Get next line, adjust for year rollover and hint datetime format."""
        # use readline here because next() iterator uses internal readahead
        # buffer so seek position is wrong
        line = self.filehandle.readline()
        **line = line.decode('utf-8', 'replace')**
        if line == '':
            raise StopIteration
        line = line.rstrip('\n')
        le = LogEvent(line)
```

But I am wondering why no one has faced the issue so far , did no one tried with python3 or it is my environment somehow. 

Thanks,
Uday

**Comment by stennie** [comment]
@udayjalagam There was attempted double-decoding in some scenarios (but not all). This was previously reported as #658.

The patch in d89555b should fix this and will be included in the 1.6.0 release later this month.

Regards,
Stennie
